pkg/skaffold/runner/runcontext/v2/context.go

```diff
@@ -198,6 +200,35 @@ func (rc *RunContext) BuildConcurrency() int                         { return rc
 func (rc *RunContext) IsMultiConfig() bool                           { return rc.Pipelines.IsMultiPipeline() }
 func (rc *RunContext) GetRunID() string                              { return rc.RunID }
 
+// GetRenderOutputPath points to the directory where the manifest rendering happens. By default, it is set to "<WORKDIR>/.kpt-pipeline".
+func (rc *RunContext) GetRenderOutputPath() (string, error) {
+	var hydratedDir string
+	var err error
+	if rc.Opts.RenderOutput == "" {
+		// Workdir is set to pwd and shall always exist.
+		hydratedDir = filepath.Join(rc.WorkingDir, constants.DefaultHydrationDir)
+	} else if hydratedDir, err = filepath.Abs(rc.Opts.RenderOutput); err != nil {
+		return "", err
+	}
+	if _, err := os.Stat(hydratedDir); os.IsNotExist(err) {
+		logrus.Infof("render output path does not exist, creating %v\n", hydratedDir)
+		if err := os.MkdirAll(hydratedDir, os.ModePerm); err != nil {
+			return "", err
+		}
+	}
+	return hydratedDir, nil
+}
+
+// GetRenderConfig returns the top tier RenderConfig.
+func (rc *RunContext) GetRenderConfig() *latestV2.RenderConfig {
+	p := rc.GetPipelines()
+	if len(p) > 0 {
+		// TODO: support multi-config
```

Comment:
```suggestion
		// TODO(yuwenma)[07/23/21]: support multi-config
```

I might also move this TODO into the method description to make it more clear why we're choosing the first config from the list.

Reply:
Good point. Done